src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -239,12 +246,22 @@ subroutine ocn_tracer_hmix_Redi_tend(meshPool, layerThickness, layerThicknessEdg
                              /(zMid(k, cell1) - zMid(k + 1, cell1)) &
                              + slopeTriadDown(k, 2, iEdge)*(tracers(iTr, k, cell2) - tracers(iTr, k + 1, cell2)) &
                              /(zMid(k, cell2) - zMid(k + 1, cell2)))
+               if (minLevelCell(cell1) < minLevelEdgeBot(iEdge)) then
+                  flux_term2 = flux_term2 + coef*RediKappa(iEdge)* &
+                               layerThicknessEdge(k, iEdge)*0.25_RKIND*kappaRediEdgeVal*slopeTriadUp(k, 1, iEdge)* &
+                               (tracers(iTr, k - 1, cell1) - tracers(iTr, k, cell1))/(zMid(k - 1, cell1) - zMid(k, cell1))
```

Comment:
```suggestion
                 k = minLevelCell(cell1)
                  flux_term2 = flux_term2 + coef*RediKappa(iEdge)* &
                               layerThicknessEdge(k, iEdge)*0.25_RKIND*kappaRediEdgeVal*slopeTriadUp(k+1, 1, iEdge)* &
                               (tracers(iTr, k, cell1) - tracers(iTr, k+1, cell1))/(zMid(k, cell1) - zMid(k+1, cell1))
```

Reply:
@mark-petersen could you check this?  I'm not sure this suggestion is exactly right.

Reply:
@vanroekel Thanks for catching this. I think we want k to be set back to `minLevelEdgeBot` as it was above before I added the reassignment to 1. And then the `k-1` index will never be 0 because we never allow `minLevel*` to be 0.

Reply:
Ah yes, that is a much better fix.

Reply:
@vanroekel @maltrud I haven't come up with a good way to have Redi work correctly for the `minLevel* != 1` cases while preserving bit-for-bit results in MPAS Standalone QU240/bgc_ecosys_test. Commit 
19dc7d9 preserves bit-for-bit results in bgc_ecosys_test but is incompatible with `minLevel* != 1` while commit https://github.com/cbegeman/MPAS-Model/commit/71663bf5ac71d361c5b03f3891851b857b62e401 is compatible with `minLevel* != 1` and introduces non-bfb results in bgc_ecosys_test. The differences are small and affect only the tracers shown below. Both branches pass all the E3SM tests [here](https://github.com/MPAS-Dev/MPAS-Model/pull/840#issuecomment-813427479). 

Let me know what you think. An alternative could be duplicating the block of code that solves for `flux_term2` but that's not very readable. 

Thanks!

```
Beginning variable comparisons for all time levels of field 'spChl'. Note any time levels reported are 0-based.
    Pass thresholds are:
       L1: 0.00000000000000e+00
       L2: 0.00000000000000e+00
       L_Infinity: 0.00000000000000e+00
0:  l1: 4.37707780575496e-23  l2: 2.42434975903054e-19  linf: 2.16840434497101e-19
 ** FAIL Comparison of spChl between forward/output.nc and
    /home/ac.cbegeman/MPAS-Ocean-output/develop_i18/ocean/global_ocean/QU240/bgc_ecosys_test/forward/output.nc
Beginning variable comparisons for all time levels of field 'spC'. Note any time levels reported are 0-based.
    Pass thresholds are:
       L1: 0.00000000000000e+00
       L2: 0.00000000000000e+00
       L_Infinity: 0.00000000000000e+00
0:  l1: 7.47021278848847e-21  l2: 5.55111512312578e-17  linf: 5.55111512312578e-17
 ** FAIL Comparison of spC between forward/output.nc and
    /home/ac.cbegeman/MPAS-Ocean-output/develop_i18/ocean/global_ocean/QU240/bgc_ecosys_test/forward/output.nc
Beginning variable comparisons for all time levels of field 'diatFe'. Note any time levels reported are 0-based.
    Pass thresholds are:
       L1: 0.00000000000000e+00
       L2: 0.00000000000000e+00
       L_Infinity: 0.00000000000000e+00
0:  l1: 7.12415007447097e-27  l2: 5.29395592033938e-23  linf: 5.29395592033938e-23
 ** FAIL Comparison of diatFe between forward/output.nc and
    /home/ac.cbegeman/MPAS-Ocean-output/develop_i18/ocean/global_ocean/QU240/bgc_ecosys_test/forward/output.nc
Beginning variable comparisons for all time levels of field 'phaeoChl'. Note any time levels reported are 0-based.
    Pass thresholds are:
       L1: 0.00000000000000e+00
       L2: 0.00000000000000e+00
       L_Infinity: 0.00000000000000e+00
0:  l1: 1.02131815467616e-22  l2: 4.47028007668187e-19  linf: 3.25260651745651e-19
 ** FAIL Comparison of phaeoChl between forward/output.nc and
    /home/ac.cbegeman/MPAS-Ocean-output/develop_i18/ocean/global_ocean/QU240/bgc_ecosys_test/forward/output.nc
```